packages/test/src/test-otel.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Test the various states of a Worker.
+ * Most tests use a mocked core, some tests run serially because they emit signals to the process
+ */
+import test from 'ava';
+import { Core, DefaultLogger, Worker } from '@temporalio/worker';
+import { CollectorTraceExporter } from '@opentelemetry/exporter-collector-grpc';
+import * as activities from './activities';
+import * as opentelemetry from '@opentelemetry/sdk-node';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import * as workflows from './workflows';
+import { WorkflowClient } from '@temporalio/client';
+
+// Un-skip this test and run it by hand to inspect outputted traces
+test.skip('Otel spans connected', async (t) => {
+  const oTelUrl = 'grpc://localhost:4317';
+  const exporter = new CollectorTraceExporter({ url: oTelUrl });
+  const otel = new opentelemetry.NodeSDK({
+    resource: new opentelemetry.resources.Resource({
+      [SemanticResourceAttributes.SERVICE_NAME]: 'node-test-otel-worker',
+    }),
+    traceExporter: exporter,
+  });
+  await otel.start();
+
+  const logger = new DefaultLogger('DEBUG');
+  // Use forwarded logging from core
+  await Core.install({
+    logger,
+    telemetryOptions: {
+      oTelCollectorUrl: oTelUrl,
+      tracingFilter: 'temporal_sdk_core=DEBUG',
+      logForwardingLevel: 'INFO',
+    },
+  });
+  const worker = await Worker.create({
+    workflowsPath: `${__dirname}/workflows`,
+    activities,
+    nodeModulesPath: `${__dirname}/../../../node_modules`,
+    taskQueue: 'test-otel',
+  });
+  const runPromise = worker.run();
+  // Catch the error here to avoid unhandled rejection
+  runPromise.catch((err) => {
+    console.error('Caught error while worker was running', err);
+  });
+
+  const client = new WorkflowClient(undefined, { workflowDefaults: { followRuns: false } });
+  const workflow = client.createWorkflowHandle(workflows.cancelFakeProgress, { taskQueue: 'test-otel' });
+  await workflow.execute();
+
+  worker.shutdown();
+  await runPromise;
```

Comment:
In newer tests I started doing this instead to avoid hanging:
```suggestion
  await Promise.all([
    workflow.execute().finally(() => worker.shutdown()),
    worker.run(),
  ])
```